packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts

```diff
@@ -190,27 +193,50 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
     },
   });
 
+  const teamMemberships = user.teams.map((membership) => ({
+    teamId: membership.team.id,
+    membershipRole: membership.role,
+  }));
+
   eventTypeGroups = ([] as EventTypeGroup[]).concat(
     eventTypeGroups,
-    user.teams.map((membership) => ({
-      teamId: membership.team.id,
-      membershipRole: membership.role,
-      profile: {
-        name: membership.team.name,
-        image: `${CAL_URL}/team/${membership.team.slug}/avatar.png`,
-        slug: membership.team.slug ? "team/" + membership.team.slug : null,
-      },
-      metadata: {
-        membershipCount: membership.team.members.length,
-        readOnly: membership.role === MembershipRole.MEMBER,
-      },
-      eventTypes: membership.team.eventTypes
-        .map(mapEventType)
-        .filter((evType) => evType.userId === null || evType.userId === ctx.user.id)
-        .filter((evType) =>
-          membership.role === MembershipRole.MEMBER ? evType.schedulingType !== SchedulingType.MANAGED : true
-        ),
-    }))
+    user.teams
+      .filter((mmship) => {
+        const metadata = teamMetadataSchema.parse(mmship.team.metadata);
+        return !metadata?.isOrganization;
+      })
+      .map((membership) => {
+        const orgMembership = teamMemberships.find(
+          (teamM) => teamM.teamId === membership.team.parentId
+        )?.membershipRole;
```

Comment:
In case we are in the presence of a sub-team belonging to an organization, we get the membership of the user in the organization as the user may be an owner and that has precedence over sub-team membership.